src/site/apt/toolchains/discovery.apt.vm

```diff
@@ -0,0 +1,148 @@
+
+   * <<<current>>>: set to <<<true>>> if this is the running JDK
+
+   * <<<lts>>>: set to <<<true>>> if JDK version is a long-term supported version
+
+   * <<<env>>>: set to the comma separated list of <<<JAVA\{xyz\}_HOME>>>> matching environment variables
+
+
+  The <<<select-jdk-toolchain>>> goal discovering and selecting a matching JDK.
+  The config below allows using the current JDK, or any other discovered JDK >= 17.
+  The benefit is that the current JDK can be kept for speed, but ensuring the usage of any JDK 17 or higher if
+  the current jdk is below the requirements.
+
++---+
+<properties>
+  <toolchain.jdk.version>[17,)</toolchain.jdk.version>
+<properties>
+
+<plugin>
+  <groupId>org.apache.maven.plugins</groupId>
+  <artifactId>maven-toolchains-plugin</artifactId>
+  <version>${project.version}</version>
+  <executions>
+    <execution>
+      <goals>
+        <goal>select-jdk-toolchain</goal>
+      </goals>
+    </execution>
+  </executions>
+</plugin>
++---+
+
+  If you use environment variables to configure your JDKs, you can use the following configuration to select
+  the toolchain which is configured using the <<<JAVA17_HOME>>> environment variable.
+
++---+
+<properties>
+  <toolchain.jdk.version>JAVA17_HOME</toolchain.jdk.version>
+<properties>
++---+
+
+* Selection mechanism
+
+  Several properties can be used to express requirements to match against discovered JDK toolchains:
+
+   * <<<version>>> / <<<toolchain.jdk.version>>>: a version range such as <<<[17,18)>>> to match against the JDK version
+
+   * <<<runtimeName>>> / <<<toolchain.jdk.runtime.name>>>
+
+   * <<<runtimeVersion>>> / <<<toolchain.jdk.runtime.version>>>
+
+   * <<<vendor>>> / <<<toolchain.jdk.vendor>>>
+
+   * <<<env>>> / <<<toolchain.jdk.env>>>: the name of an environment variable that the JDK toolchain must match
+
+  The <<<useJdk>>> can be used to define whether the current JDK can be used if it matches the requirements.
+
+* Sorting
+
+  Multiple discovered JDK toolchains may match the above requirements.  In such a case, you may want to express
+  preferences to use to sort the toolchains.  This can be done using the <<<comparator>>> configuration which is a
+  comma separated list of criterions amongst the following:
+
+   * <<<lts>>>: prefer LTS toolchains
+
+   * <<<current>>>: prefer the current JDK
+
+   * <<<env>>>: prefer toolchains discovered from environment variables
+
+   * <<<version>>>: prefer higher JDK versions
+
+   * <<<vendor>>>: sort alphabetically by vendor name
+
+  The default value is <<<lts,current,env,version,vendor>>>.
+
+* Toolchains XML file
+
+  The generation of the <<<toolchains.xml>>> file is not necessary to use discovered toolchains.
+  The <<<select-jdk-toolchain>>> will select a toolchain amongst explicitly configured toolchains and discovered
+  toolchains. The information for discovered toolchains are cached in <<<~/.m2/discovered-toolchains-cache.xml>>> file
```

Comment:
The information for discovered --> Discovered

Reply:
That's not completely right, as the discovery is two steps: find JDKs, gather info for each JDK.  Only the second step is cached.  For in order to simplify the doc, I'll go with your proposal.